Shared/Samples/Add items to portal/AddItemsToPortalView.swift

```diff
@@ -0,0 +1,269 @@
+    /// - Parameter portalItem: The portal item to delete.
+    private func deletePortalItem(_ portalItem: PortalItem) async throws {
+        try await portalUser!.delete(portalItem)
+        
+        // Refreshes the portal items list after deletion.
+        try await fetchPortalItems()
+    }
+    
+    /// Fetches the portal items from a user.
+    private func fetchPortalItems() async throws {
+        if let portalUser {
+            portalItems = try await portalUser.content.items.sorted {
+                ($0.creationDate ?? .distantPast) > ($1.creationDate ?? .distantPast)
+            }
+        }
+    }
+    
+    /// Signs out from the portal by revoking OAuth tokens and clearing credential stores.
+    private func signOut() async {
+        await ArcGISEnvironment.authenticationManager.revokeOAuthTokens()
+        await ArcGISEnvironment.authenticationManager.clearCredentialStores()
+    }
+    
+    /// Sets up new ArcGIS and Network credential stores that will be persisted in the keychain.
+    private func setupPersistentCredentialStorage() async throws {
+        try await ArcGISEnvironment.authenticationManager.setupPersistentCredentialStorage(
+            access: .whenUnlockedThisDeviceOnly,
+            synchronizesWithiCloud: false
+        )
+    }
+}
+
+/// A view that displays information about a portal item for viewing in a list.
+struct PortalItemView: View {
+    /// The portal item to display information about.
+    let item: PortalItem
+    
+    /// The size of the thumbnail.
+    private let thumbnailSize: CGFloat = 64
+    
+    /// The thumbnail image of the map area.
+    @State private var thumbnail: UIImage?
+    
+    var body: some View {
+        HStack {
+            if let thumbnail {
+                Image(uiImage: thumbnail)
+                    .resizable()
+                    .aspectRatio(contentMode: .fill)
+                    .frame(width: thumbnailSize, height: thumbnailSize)
+                    .clipShape(.rect(cornerRadius: 10))
+            } else {
+                Image(systemName: "map")
+                    .imageScale(.large)
+                    .foregroundStyle(.secondary)
+                    .frame(width: thumbnailSize, height: thumbnailSize)
+                    .background(Color(uiColor: UIColor.systemGroupedBackground))
+                    .clipShape(.rect(cornerRadius: 10))
+            }
+            VStack(alignment: .leading, spacing: 2) {
+                Text(item.title.replacingOccurrences(of: "_", with: " ").capitalized)
+                    .font(.headline)
+                    .lineLimit(1)
+                Text("Owner: \(item.owner), Views: \(item.viewCount)")
+                    .font(.footnote)
+                Text(item.snippet)
+                    .font(.caption2)
+                    .foregroundColor(.secondary)
+                    .lineLimit(2)
+            }
+        }
+        .task {
+            try? await item.thumbnail?.load()
```

Comment:
I never seem to get the thumbnail. Looks like this throws the following error. Expected?

> ArcGIS.IllegalStateError(details: "Loadable image doesn\'t have a URI.")

Reply:
Let me ask around to see if this is a known issue…

Reply:
Haven't heard an confirmative response yet… will request review again once I hear back.

Reply:
Checked with [Laura Bursolo](https://www.esri.com/arcgis-blog/author/lbusolo) from AGOL who's in charge of the frontend of AGOL portal item side of things. She confirmed that the same behavior presents on devext (null thumbnail field in the JSON), and cannot tell for sure what is the expected behavior. Therefore, let's say for now it is expected.